File: frontend/src/app/instruments/models/enums.ts
/**
 * Enums for instruments feature
 * These match the backend enums in app/models/enums.py
 */

/**
 * Valid instrument types
 */
export enum InstrumentType {
  EQUITY = 'Equity',
  BOND = 'Bond',
  ETF = 'ETF',
  FUTURE = 'Future'
}

/**
 * Get all instrument type values as array
 */
export function getInstrumentTypes(): string[] {
  return Object.values(InstrumentType);
}

/**
 * Valid currency codes (ISO 4217)
 */
export enum Currency {
  CHF = 'CHF',
  EUR = 'EUR',
  USD = 'USD'
}

/**
 * Get all currency values as array
 */
export function getCurrencies(): string[] {
  return Object.values(Currency);
}

File: backend/app/models/enums.py
"""
Enums for shared constants across the application.
"""

from enum import Enum


class InstrumentType(str, Enum):
    """Valid instrument types."""
    EQUITY = "Equity"
    BOND = "Bond"
    ETF = "ETF"
    FUTURE = "Future"

    @classmethod
    def values(cls):
        """Return list of valid values."""
        return [e.value for e in cls]


class Currency(str, Enum):
    """Valid currency codes (ISO 4217)."""
    CHF = "CHF"
    EUR = "EUR"
    USD = "USD"

    @classmethod
    def values(cls):
        """Return list of valid values."""
        return [e.value for e in cls]

